src/graph/graphNodes.ts

```diff
@@ -0,0 +1,128 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import * as vscode from 'vscode';
+import * as path from 'path';
+import { Command } from 'vscode';
+import { DocumentClient } from 'documentdb';
+import { INode } from '../nodes';
+import * as util from "./../util";
+import { GraphViewsManager } from "./GraphViewsManager";
+import { GraphConfiguration } from './GraphConfiguration';
+import { GraphViewServer } from './GraphViewServer';
+
+export class GraphDatabaseNode implements INode {
+	public readonly contextValue: string = "cosmosGraphDatabase";
+
+	private _graphEndpoint: string;
+	private _graphPort: number;
+
+	constructor(readonly id: string, private readonly _masterKey: string, private readonly _documentEndpoint: string, private readonly _graphViewsManager: GraphViewsManager, readonly server: INode) {
+		this._parseEndpoint(_documentEndpoint);
+	}
+
+	private _parseEndpoint(documentEndpoint: string): void {
+		// Document endpoint: https://<graphname>.documents.azure.com:443/
```

Comment:
Would be simpler if you just used 'url'. Something like:
```
import * as url from 'url';
const endpointUrl = url.parse(this._documentEndpoint);
endpointUrl.hostname
endpointUrl.port
```

Reply:
Added P2 TODO